lib/reform/form/dry/input_hash.rb

```diff
@@ -0,0 +1,37 @@
+module Reform::Form::Dry
+  module InputHash
+    private
+
+    # if dry_error is a hash rather than an array then it contains
+    # the messages for a nested property
+    # these messages need to be added to the correct collection
+    # objects.
+
+    # collections:
+    # {0=>{:name=>["must be filled"]}, 1=>{:name=>["must be filled"]}}
+
+    # Objects:
+    # {:name=>["must be filled"]}
+    # simply load up the object and attach the message to it
+
+    # we can't use to_nested_hash as it get's messed up by composition.
+    def input_hash(form)
+      hash = form.class.nested_hash_representer.new(form).to_hash
+      symbolize_hash(hash)
+    end
+
+    # dry-v needs symbolized keys
+    # TODO: Don't do this here... Representers??
+    def symbolize_hash(old_hash)
+      old_hash.each_with_object({}) do |(k, v), new_hash|
```

Comment:
I think I saw some Dry-specific code somewhere in `dry-equalizer` or something to do that, maybe @solnic can help?

Reply:
@apotonick dry-validation symbolizes hashes for `Params` and `JSON` schemas

Reply:
I believe we need to keep this for old versions of dry-v?